ci/docker/runtime_functions.sh

```diff
@@ -1437,9 +1408,404 @@ deploy_docs() {
 
     export CC="ccache gcc"
     export CXX="ccache g++"
-    make docs SPHINXOPTS=-W USE_MKLDNN=0
 
+    build_python_docs
+
+    popd
+}
+
+
+build_docs_setup() {
+    build_folder="docs/_build"
+    mxnetlib_folder="/work/mxnet/lib"
+
+    mkdir -p $build_folder
+    mkdir -p $mxnetlib_folder
+}
+
+build_ubuntu_cpu_docs() {
+    set -ex
+    export CC="gcc"
+    export CXX="g++"
+    build_ccache_wrappers
+    make \
+        DEV=1                         \
+        USE_CPP_PACKAGE=1             \
+        USE_BLAS=openblas             \
+        USE_MKLDNN=0                  \
+        USE_DIST_KVSTORE=1            \
+        USE_LIBJPEG_TURBO=1           \
+        USE_SIGNAL_HANDLER=1          \
+        -j$(nproc)
+}
+
+
+build_jekyll_docs() {
+    set -ex
+    source /etc/profile.d/rvm.sh
+
+    pushd .
+    build_docs_setup
+    pushd docs/static_site
+    make clean
+    make html
+    popd
+
+    GZIP=-9 tar zcvf jekyll-artifacts.tgz -C docs/static_site/build html
+    mv jekyll-artifacts.tgz docs/_build/
+    popd
+}
+
+
+build_python_docs() {
+   set -ex
+   pushd .
+
+   build_docs_setup
+
+   pushd docs/python_docs
+   eval "$(/work/miniconda/bin/conda shell.bash hook)"
+   conda env create -f environment.yml -p /work/conda_env
+   conda activate /work/conda_env
+   pip install themes/mx-theme
+   pip install -e /work/mxnet/python --user
+
+   pushd python
+   make clean
+   make html EVAL=0
+
+   GZIP=-9 tar zcvf python-artifacts.tgz -C build/_build/html .
+   popd
+
+   mv python/python-artifacts.tgz /work/mxnet/docs/_build/
+   popd
+
+   popd
+}
+
+
+build_c_docs() {
+    set -ex
+    pushd .
+
+    build_docs_setup
+    doc_path="docs/cpp_docs"
+    pushd $doc_path
+
+    make clean
+    make html
+
+    doc_artifact="c-artifacts.tgz"
+    GZIP=-9 tar zcvf $doc_artifact -C build/html/html .
+    popd
+
+    mv $doc_path/$doc_artifact docs/_build/
+
+    popd
+}
+
+
+build_r_docs() {
+    set -ex
+    pushd .
+
+    build_docs_setup
+    r_root='R-package'
+    r_pdf='mxnet-r-reference-manual.pdf'
+    r_build='build'
+    docs_build_path="$r_root/$r_build/$r_pdf"
+    artifacts_path='docs/_build/r-artifacts.tgz'
+
+    mkdir -p $r_root/$r_build
+
+    unittest_ubuntu_minimal_R
+
+    pushd $r_root
+
+    R_LIBS=/tmp/r-site-library R CMD Rd2pdf . --no-preview --encoding=utf8 -o $r_build/$r_pdf
+
+    popd
+
+    GZIP=-9 tar zcvf $artifacts_path $docs_build_path
```

Comment:
what does this line do ?

Reply:
this zips up with high compression (GZIP=-9) the generated artifact in a specific path so that it can then be picked up by CI for building the entire website as the sum of its components